freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UmsCredentialProvider.java

```diff
@@ -0,0 +1,28 @@
+package com.sequenceiq.freeipa.service.freeipa.user;
+
+import com.cloudera.thunderhead.service.usermanagement.UserManagementProto.GetActorWorkloadCredentialsResponse;
+import com.sequenceiq.cloudbreak.auth.altus.GrpcUmsClient;
+import com.sequenceiq.freeipa.service.freeipa.user.model.WorkloadCredential;
+import org.springframework.stereotype.Component;
+
+import javax.inject.Inject;
+import java.time.Instant;
+import java.util.Optional;
+
+import static com.sequenceiq.cloudbreak.auth.altus.GrpcUmsClient.INTERNAL_ACTOR_CRN;
+
+@Component
+public class UmsCredentialProvider {
+
+    @Inject
+    private GrpcUmsClient grpcUmsClient;
+
+    public WorkloadCredential getCredentials(String userCrn, Optional<String> requestId) {
```

Comment:
I could swear that I've seen this code before.

Reply:
Well this was extracted off of UmsUsersStateProvider, and thanks to your comment I just realized I forgot remove it from there since it's no longer used (weird that code quality doesn't complain about unused functions).